pkg/apis/numaflow/v1alpha1/pipeline_types.go

```diff
@@ -509,6 +524,37 @@ func (wm Watermark) GetMaxDelay() time.Duration {
 	return time.Duration(0)
 }
 
+func (wm Watermark) GetIdleSource() *IdleSource {
```

Comment:
What is the point of this function? Isn't it the same as `return wm.IdleSource`?

Reply:
yaa, it's same as `wm.IdleSource` also it's not used anywhere, I'll remove it